test/server/cards/02_SHD/leaders/FinnThisIsARescue.spec.ts

```diff
@@ -0,0 +1,127 @@
+describe('Finn, This is a Rescue', function () {
+    integration(function (contextRef) {
+        describe('Finn\'s undeployed ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['jedi-lightsaber'],
+                        leader: { card: 'finn#this-is-a-rescue', deployed: false },
+                        groundArena: ['battlefield-marine'],
+                        resources: 5
+                    },
+                    player2: {
+                        hand: ['top-target'],
+                        groundArena: ['wampa'],
+                        resources: 5
+                    }
+                });
+            });
+
+            it('should defeat a friendly upgrade and give a shield token', function () {
+                const { context } = contextRef;
+
+                // Equip a friendly upgrade to battlefield marine
+                context.player1.clickCard(context.jediLightsaber);
```

Comment:
You can remove this and have the test start with the lightsaber equipped. See my above comment.